pf4j/src/main/java/org/pf4j/AbstractPluginManager.java

```diff
@@ -331,6 +331,9 @@ protected boolean unloadPlugin(String pluginId, boolean unloadDependents) {
             }
         }
 
+        // resolve the plugins again (update plugins graph)
+        resolveDependencies();
```

Comment:
Here, I have some uncertainties regarding performance (the case in which a recursive upload is made => resolveDependencies method is called several times)

Reply:
@decebals Can we update the dependencies graph by providing a method similar to `void removePlugin (PluginDescriptor descriptor)` in DependencyResolver

Reply:
@guqing I know that I tried to do this, but it is not trivial, I have encountered more complex scenarios. I have some modifications on a local branch.
By the way, did you have the opportunity to test the fix from this PR? It is not optimal in terms of performance, but I don't think that performance is a problem in the given scenario.

Reply:
Hi @decebals ，After testing it on my application, I found it works very well. I appreciate you taking the time to address this issue ♥️

Reply:
@guqing I added a new commit to this PR that eliminate the possible performance penalty.
Please check this PR in this form (with this latest commit). If it's OK I will release a new version that includes this fix.

Reply:
Hi @decebals , testing with the latest commit still works very well 👍